src/sth_database.js

```diff
@@ -425,6 +425,109 @@
+   * @param results The array of results
+   * @param resolution The resolution
+   * @param from The starting date
+   * @param to The ending date
+   * @param aggregatedFunction The aggregation function
+   * @param shouldFilter Flag indicating if null results should be filtered
+   */
+  function filterResults(results, resolution, from, to, aggregatedFunction, shouldFilter) {
+    if (results.length) {
+      var points;
+      if (from &&
+        results[0]._id.origin.getTime() === sthHelper.getOrigin(from, resolution).getTime()) {
+        points = results[0].points;
+        var minOffset;
+        switch (resolution) {
+          case sthConfig.RESOLUTION.SECOND:
+            minOffset = from.getUTCSeconds();
+            break;
+          case sthConfig.RESOLUTION.MINUTE:
+            minOffset = from.getUTCMinutes();
+            break;
+          case sthConfig.RESOLUTION.HOUR:
+            minOffset = from.getUTCHours();
+            break;
+          case sthConfig.RESOLUTION.DAY:
+            minOffset = from.getUTCDate();
+            break;
+          case sthConfig.RESOLUTION.MONTH:
+            minOffset = from.getUTCMonth();
+            break;
+        }
+        for (var i = 0; i < points.length; i++) {
+          if (points[i] && points[i].samples && points[i].offset < minOffset) {
+            if (shouldFilter) {
+              points.splice(i, 1);
+              i--;
+            } else {
+              points[i].samples = 0;
+              if (aggregatedFunction === 'occur') {
+                points[i].occur = {};
+              } else if (aggregatedFunction === 'min') {
+                points[i].min = Number.POSITIVE_INFINITY
+              } else if (aggregatedFunction === 'max') {
+                points[i].max = Number.NEGATIVE_INFINITY;
+              } else {
+                points[i][aggregatedFunction] = 0;
+              }
+            }
+          }
+        }
+        if (!points.length) {
+          results.splice(0, 1);
+        }
+      }
+      if (results.length && to &&
+        results[results.length - 1]._id.origin.getTime() === sthHelper.getOrigin(to, resolution).getTime()) {
+        points = results[results.length - 1].points;
+        var maxOffset;
+        switch (resolution) {
+          case sthConfig.RESOLUTION.SECOND:
+            maxOffset = to.getUTCSeconds();
+            break;
+          case sthConfig.RESOLUTION.MINUTE:
+            maxOffset = to.getUTCMinutes();
+            break;
+          case sthConfig.RESOLUTION.HOUR:
+            maxOffset = to.getUTCHours();
+            break;
+          case sthConfig.RESOLUTION.DAY:
+            maxOffset = to.getUTCDate();
+            break;
+          case sthConfig.RESOLUTION.MONTH:
+            maxOffset = to.getUTCMonth();
+            break;
+        }
+        for (var i = points.length - 1; i >= 0; i--) {
+          if (points[i] && points[i].samples && points[i].offset > maxOffset) {
```

Comment:
The same than above, but applied to the maximum offset. In this case the (sorted?) list must be iterated starting from the last element.